src/cpp/mip/metadata/mip_meta_utils.hpp

```diff
@@ -151,6 +151,45 @@ constexpr size_t serializedSizeForBasicType(Type type, const void* info=nullptr)
 }
 constexpr size_t serializedSizeForBasicType(const TypeInfo& type) { return serializedSizeForBasicType(type.type, type.infoPtr); }
 
+///@brief Gets the size of a basic type (including bitfields and enums if class_ is not NULL).
+///
+constexpr size_t serializedAlignForBasicType(Type type, const void* info=nullptr)
+{
+    switch(type)
+    {
+    case Type::CHAR:
```

Comment:
Technically these should all be their own cases:
```
case Type::CHAR: return alignof(char);
case Type::BOOL: return alignof(bool);
case Type::U8:   return alignof(uint8_t);
```
etc. In particular, float and double shouldn't be grouped with the integer types.